spring-client-common/src/main/java/io/camunda/commons/auth/SaaSAuthentication.java

```diff
@@ -0,0 +1,104 @@
+package io.camunda.commons.auth;
+
+import org.apache.hc.client5.http.classic.methods.HttpPost;
+import org.apache.hc.client5.http.impl.classic.CloseableHttpClient;
+import org.apache.hc.client5.http.impl.classic.CloseableHttpResponse;
+import org.apache.hc.client5.http.impl.classic.HttpClients;
+import org.apache.hc.core5.http.io.entity.EntityUtils;
+import org.apache.hc.core5.http.io.entity.StringEntity;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.lang.invoke.MethodHandles;
+import java.util.AbstractMap;
+import java.util.HashMap;
+import java.util.Map;
+
+import static io.camunda.commons.http.JsonUtils.toJson;
+import static io.camunda.commons.http.JsonUtils.toResult;
+
+public class SaaSAuthentication extends JwtAuthentication {
+
+  private static final Logger LOG = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+
+  private String authUrl;
+  private JwtConfig jwtConfig;
+  private Map<Product, String> tokens;
+  private Map<Product, Integer> expirations;
+
+  public SaaSAuthentication() {
+    tokens = new HashMap<>();
+    expirations = new HashMap<>();
+  }
+
+  public SaaSAuthentication jwtConfig(JwtConfig jwtConfig) {
+    this.jwtConfig = jwtConfig;
+    return this;
+  }
+
+  @Override
+  public Authentication build() {
+    authUrl = "https://login.cloud.camunda.io/oauth/token";
+    jwtConfig.getMap().forEach(this::retrieveToken);
+    return this;
+  }
+
+  private void retrieveToken(Product product, Credential credential) {
+    try {
+      HttpPost httpPost = new HttpPost(authUrl);
+      httpPost.addHeader("Content-Type", "application/json");
+      TokenRequest tokenRequest = new TokenRequest(getAudience(product), credential.clientId, credential.clientSecret);
+
+      httpPost.setEntity(new StringEntity(toJson(tokenRequest)));
+      CloseableHttpClient client = HttpClients.createDefault();
```

Comment:
We could re-use an existing client here instead of creating a new one for every token request.

Reply:
Addressed in https://github.com/camunda-community-hub/spring-zeebe/pull/510/commits/4036ec7c35e84168e582d91e2ff44edec6d50240